### DISCUSSION TITLE
Error using prior layer weights

### DISCUSSION THREAD

**Discussion by dskhanirfan** [FETS-AI/Challenge#48] [General]
If I use the following available code to access the prior layer weights in the notebook, I get an error, How can I fix it?

```python
for record in db_iterator:
        if (
            record['round'] == (fl_round - 1)
            and record['tensor_name'] == tensor_name
            and 'aggregated' in record['tags']
            and 'delta' not in record['tags']
           ):
            previous_tensor_value = record['nparray']
            break
```
**Error**
```python-traceback
File ".py_venv_3.7.3/lib/python3.7/site-packages/openfl/pipelines/tensor_codec.py", line 179, in generate_delta
    return delta_tensor_key, nparray - base_model_nparray
TypeError: unsupported operand type(s) for -: 'NoneType' and 'float'
srun: error: r03g01: task 0: Exited with exit code 1
```

**Comment by sarthakpati** [comment]
Hmm, I don't believe I recognize the function. Pinging @brandon-edwards, @msheller, @psfoley and @alexey-gruzdev for clarification.

**Reply by alexey-gruzdev** [reply]
@dskhanirfan could you please provide a full traceback and exact modifications that you made in the notebook for us to investigate?

**Reply by sarthakpati** [reply]
@dskhanirfan if you are uncomfortable posting modifications here, please email us at challenge@fets.ai and please reference this thread.
